src/main/java/TaskList.java

```diff
@@ -0,0 +1,290 @@
+import java.io.IOException;
+import java.util.List;
+
+public class TaskList {
+    private List<Task> storingList;
+    Storage storage;
+
+    /**
+     * Constructor for TaskList.
+     *
+     * @param load    the arrayList obtained from Storage class's load method. Contains task from hard disk.
+     * @param storage Storage class previously created from Duke.
+     */
+    public TaskList(List<Task> load, Storage storage) {
+        //contains the task list e.g., it has operations to add/delete tasks in the list
+        storingList = load;
+        this.storage = storage;
+    }
```

Comment:
If possible, consider refactoring Storage outside of TaskList. That way you can reuse these two classes in other applications you program.

Reply:
Thanks for the suggestion! Will implement it this way later.